### PR TITLE
Features

### PR DESCRIPTION
**Pull request by ericksaga**
Add/Remove asteroids
Collisions
Debugging mode

### CODE REVIEW

File: asteroids_code/BoilerPlate/Asteroid.cpp [hunk 1]
```diff
@@ -55,60 +89,142 @@ void Asteroid::Render()
 	glEnd();
 }
 
-void Asteroid::MoveForward()
+void Asteroid::MoveForward(const float delta_time)
 {
-	float angle_radian = math_master.DegreeToRadian(angle_degree);
-	origin += Vector2(-ASTEROID_SPEED * sin(angle_radian), ASTEROID_SPEED * cos(angle_radian));
+	//calculate speed depending on the size, smallest = fastest
+	float LC_angle_radian = math_master.DegreeToRadian(angle_degree);
```
Comment (rroa) [inline asteroids_code/BoilerPlate/Asteroid.cpp]: What's LC?

File: asteroids_code/BoilerPlate/Asteroid.cpp [hunk 2]
```diff
@@ -55,60 +89,142 @@ void Asteroid::Render()
...
-	if (auto_rotate > 360)
-	{
-		auto_rotate -= 360;
-	}
-	Warp(w_width, w_height);
+	Entity::Update(w_width, w_height, delta_time);
+	SizeRefactor();
+}
 
+void Asteroid::SizeRefactor()
```
Comment (rroa) [inline asteroids_code/BoilerPlate/Asteroid.cpp]: In this context, what does refactor mean?

File: asteroids_code/BoilerPlate/Asteroid.cpp [hunk 3]
```diff
@@ -55,60 +89,142 @@ void Asteroid::Render()
...
 		asteroid_state = MEDIUM;
 	}
 }
 
 int Asteroid::GetSize()
 {
 	return asteroid_state;
+}
+
+void Asteroid::AssignOrientation(int random_number)
```
Comment (rroa) [inline asteroids_code/BoilerPlate/Asteroid.cpp]: I believe you mean set.

File: asteroids_code/BoilerPlate/Asteroid.cpp [hunk 4]
```diff
@@ -55,60 +89,142 @@ void Asteroid::Render()
...
+{
+	angle_degree = (float) (random_number % 360);
+}
+
+void Asteroid::AssignPosition(int random_number_x, int random_number_y)
+{
+	/*assing the random position with the values, for it to be more random if the number is even the
+	position is negative, and if it's odd it would be positive*/
+	float LC_pos_x, LC_pos_y;
+	LC_pos_x = (float)(random_number_x % 201 + 100);
```
Comment (rroa) [inline asteroids_code/BoilerPlate/Asteroid.cpp]: Magic numbers! Please move these out!

File: asteroids_code/BoilerPlate/Bullet.cpp [hunk 5]
```diff
@@ -0,0 +1,74 @@
+#include "Bullet.hpp"
+//OpenGL include
+#include <GL/glew.h>
+#include <SDL2/SDL_opengl.h>
+
+Bullet::Bullet()
+{
+	origin = Vector2(0.0f, 0.0f);
```
Comment (rroa) [inline asteroids_code/BoilerPlate/Bullet.cpp]: You should have a static function in `Vector2` that is called Origin already. In addition, the default constructor should return a vector on the origin.

File: asteroids_code/BoilerPlate/Bullet.cpp [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+void Bullet::Render()
+{
+	glLoadIdentity();
+	glTranslatef(origin.x, origin.y, 0.0f);
+	glRotatef(angle_degree, 0.0f, 0.0f, 1.0f);
+	glColor3f(1.0f, 1.0f, 1.0f);
+	BulletRender();
+	if (debugging)
```
Comment (rroa) [inline asteroids_code/BoilerPlate/Bullet.cpp]: if debugging is a private member, something should differentiate it from local variables.

File: asteroids_code/BoilerPlate/App.hpp [hunk 7]
```diff
@@ -52,21 +58,41 @@ namespace Engine
 		void OnExit							( ) override;
 		void OnKeyDown						( SDL_KeyboardEvent keyBoardEvent ) override;
 		void OnKeyUp						( SDL_KeyboardEvent keyBoardEvent ) override;
-
-
+		void GenerateAsteroid				(Asteroid*);
```
Comment (rroa) [inline asteroids_code/BoilerPlate/App.hpp]: If you are "generating" an asteroid, why would you need an asteroid?

**Review by rroa — COMMENTED** [review]
Be careful when naming functions and variables, sometimes the messages does not get across and is hard to interpret your intention.

File: asteroids_code/BoilerPlate/Asteroid.cpp (hunk 1 above)
Comment (ericksaga) [reply asteroids_code/BoilerPlate/Asteroid.cpp]: it s a convention to differentiate local variables from others variables meaning Local

**Pull request merged**